### ISSUE TITLE
[Blazor] Some of the default UI styling is not accessible in dark mode

### ISSUE THREAD

**Issue by MackinnonBuck** [dotnet/aspnetcore#57111]
There are cases where the styling of some of Blazor's built-in UI elements gets adversely affected by global dark mode styling.

Some examples:

1. The server reconnect UI:
![Blazor server reconnect UI](https://github.com/user-attachments/assets/a7951f26-0e50-49f6-b176-e77d2b5d1c99)

2. The "Yellow bar":
![Yellow bar](https://github.com/user-attachments/assets/d76e0a4a-167e-4c5c-aab2-f4b0e90fb9bb)

We should consider styling these elements in ways that don't inherit from other styles defined in the app.

**Comment by javiercn** [comment]
Looked at the four pieces of UI we produce:
* Blazor Error UI (yellow bar) needs `color-scheme: light only` on the template to avoid issues with auto dark mode.
* Blazor Loading animation: Seems fine, no changes needed.
* Error boundary UI: It's fine, no changes needed.
* Blazor Server reconnect UI. From what was described in this issue, there were problems when styles where changes. To avoid that, I've put all the content inside a shadow root to avoid accidental styling.
 * ![image](https://github.com/user-attachments/assets/7d05b115-e2c7-4cc9-91fc-3123874b9465)
 * ![image](https://github.com/user-attachments/assets/85c632e6-c0de-4e05-9d78-4308a4707052)
 * ![image](https://github.com/user-attachments/assets/8c85998e-b4b5-45db-acf9-baf8ba0c0998)
   * Contrast in auto dark mode is not great, but the text is clear.
